WordPress/gutenberg
# Site Editor: show pattern category step in navigation for mobile

## What?

So that users can access template parts in mobile view from the patterns panel, show the patterns category list.

Props to @t-hamano for the nudge.

See: https://github.com/WordPress/gutenberg/issues/67460#issuecomment-2513461900


> [!NOTE]
> There is a known inconsistency in the WP logo navigation: when you enter the data view for a patterns category, the WP logo takes you back to the main Design menu (not back a step).


## Why?

In themes such as TT4, the "All patterns" view won't show the template parts and are therefor inaccessible.

## How?

Checking for postType and categoryId. If it's there, show `<PagePatterns />`.

## Testing Instructions

1. Switch to TT4
2. In mobile view (narrow the viewport width), head to Appearance > Editor
3. Now click on "Patterns"
4. You should see the categories list. 
5. Click on template parts and patterns, make sure navigating works and that you can enter the editor from the dataview list.


## Screenshots or screencast 

### Before

https://github.com/user-attachments/assets/174de99d-d931-4f24-9edf-b6e9f5a59f0f


#### After


https://github.com/user-attachments/assets/c2b22ab6-f442-48e2-860f-5c260b0ea2f7
